### PR TITLE
Update ref-based RNA-seq tutorial

### LINKED ISSUE #531: replace DAVID tool with an alternative
Hi,
Trying to run the ref-based RNA-seq in a GVL instance.
Can't find DAVID tool (Mentioned on page 21) in tool-shed - what should I be looking for?

TIA,
Mark

### LINKED ISSUE #552: update tutorials using HISAT2 to new version
Creating this issue to flag that, as HISAT2 has been updated, the current screenshots and text in some GTN tutorials no longer match the newest version of the tool, so will need to be updated at some stage e.g. these tuts

- https://galaxyproject.github.io/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html
- https://galaxyproject.github.io/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html
- https://galaxyproject.github.io/training-material/topics/transcriptomics/tutorials/srna/tutorial.html

Happy to help update if you like when I get a chance.

### PR DESCRIPTION
**Pull request by bebatut**
- [X] Change datasets
- [X] Replace HISAT by STAR (should also fix #552)
- [X] Replace HTSeq-count by FeatureCount
- [X] Replace DAVID by goseq (to fix #531)
- [X] Add heatmap generation
- [X] Add more details about the mappers
- [X] Move and adapt section about the strandness
- [X] Update workflows and tools
- [X] Upload new datasets on Zenodo
- [X] Update the data library

### CODE REVIEW

File: topics/transcriptomics/tutorials/ref-based/tutorial.md [hunk 1]
```diff
@@ -407,50 +346,169 @@ The next step in the RNA-Seq data analysis is quantification of expression level
...
+![Effects of RNAseq library types](../../images/rnaseq_library_type.png "Effects of RNAseq library types (adapted from Sailfish documentation)")
+
+In practice, with Illumina paired-end RNAseq protocols, you are unlikely to uncover many of these possibilities. You will either deal with:
+
+- Unstranded RNAseq data
+- Stranded RNAseq data produced with Illumina TrueSeq RNAseq kits and [dUTP tagging](https://nar.oxfordjournals.org/content/37/18/e123) (**ISR**)
+
+This information should usually come with your FASTQ files, ask your sequencing facility! If not, try to find them on the site where you downloaded the data or in the corresponding publication.
+
+Another option is to estimate these parameters with a tool called **Infer Experiment**. This tool takes the output of mapping, subsample a subset of reads, compare their genome coordinates and strands with those of the reference gene model (from an annotation file). Based on the strand of the genes, it can gauge whether sequencing is strand-specific, and if so, how reads are stranded.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/ref-based/tutorial.md]: the output of your mappings (BAM files), takes a subsample of your reads and compares their genome coordinates and strands ...

File: topics/transcriptomics/tutorials/ref-based/tutorial.md [hunk 2]
```diff
@@ -501,46 +559,110 @@ Here treatment is the primary factor which we are interested in. The sequencing
...
+    5. Dispersion estimates: gene-wise estimates (black), the fitted values (red), and the final maximum a posteriori estimates used in testing (blue)
 
-4.	Standard error estimate for the log2 fold change estimate
-5.	[Wald](https://en.wikipedia.org/wiki/Wald_test) statistic
-6.	*p*-value for the statistical significance of this change
-7.	*p*-value adjusted for multiple testing with the Benjamini-Hochberg procedure which controls false discovery rate ([FDR](https://en.wikipedia.org/wiki/False_discovery_rate))
+        This dispersion plot is typical, with the final estimates shrunk from the gene-wise estimates towards the fitted estimates. Some gene-wise estimates are flagged as outliers and not shrunk towards the fitted value. The amount of shrinkage can be more or less than seen here, depending on the sample size, the number of coefficients, the row mean and the variability of the gene-wise estimates.
 
-> ### {% icon hands_on %} Hands-on: Analysis of the differential gene expression (2)
+- A summary files with for each gene:
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/ref-based/tutorial.md]: A summary file with the following values for each gene:

File: topics/transcriptomics/tutorials/ref-based/tutorial.md [hunk 3]
```diff
@@ -549,110 +671,191 @@ The first output of **DESeq2** is a tabular file. The columns are:
...
+> ### {% icon question %} Questions
+>
+> - Do you observe any tendency in the data?
+> - What is changing if we select `Plot the data as it is` in "Advanced - log transformation"?
+> - Can you generate an heatmap the normalized counts for the up-regulated genes with FC > 2?
+>
+> <details>
+> <summary>Click to view answers</summary>
+> <ol type="1">    
+>   <li></li>
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/ref-based/tutorial.md]: answer is missing

File: topics/transcriptomics/tutorials/ref-based/tutorial.md [hunk 4]
```diff
@@ -549,110 +671,191 @@ The first output of **DESeq2** is a tabular file. The columns are:
...
+>    - "Differentially expressed genes file" to first file generated on previous step
+>    - "Gene lengths file" to second file generated on previous step
+>    - "Gene categories" to `Get categories`
+>    - "Select a genome to use" to `Fruit fly (dm6)`
+>    - "Select Gene ID format" to `Ensembl Gene ID`
+>    - "Select one or more categories" to `GO: Cellular Component`, `GO: Biological Process`, `GO: Molecular Function`
+>
+{: .hands_on}
+
+goseq generates a big table with for each GO terms:
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/ref-based/tutorial.md]: goseq generates a big table with the following columns for each GO term:

File: topics/transcriptomics/tutorials/ref-based/tutorial.md [hunk 5]
```diff
@@ -127,168 +160,91 @@ As the genome of *Drosophila melanogaster* is known and assembled, we can use th
...
 
-In the previous illustration, you can see that in the dUTP method, for example, only the first cDNA strand, synthesized through reverse transcription of the original RNA, gets sequenced, while the second cDNA strand corresponding to the original RNA strand is degraded because of the dUTP incorporated into it.
+<p>A part of the read (blue arrow) is first mapped to the genome using the global FM index. The HISAT then tries to extend the alignment directly utilizing the genome sequence (violet arrow). In (<strong>a</strong>) it succeeds and this read aligned as it completely resides within an exon. In (<strong>b</strong>) the extension hits a mismatch. Now HISAT takes advantage of the local FM index overlapping this location to find the appropriate matting for the remainder of this read (green arrow). The (<strong>c</strong>) shows a combination these two strategies: the beginning of the read is mapped using global FM index (blue arrow), extended until it reaches the end of the exon (violet arrow), mapped using local FM index (green arrow) and extended again (violet arrow).</p>
```
Comment (nsoranzo) [inline topics/transcriptomics/tutorials/ref-based/tutorial.md]: s/matting/mapping/ I suppose

**Comment by nsoranzo** [comment]
> Replace HISAT by STAR

Doesn't this make it more difficult to run the tutorial on low-end computers and Docker images?

**Comment by bgruening** [comment]
@nsoranzo I don't have hard numbers, but what is really expensive is generating the star index. The mapping is okish, especially with dm6.

**Comment by nekrut** [comment]
@bebatut I think this is great. We need to get rid of the old reference-based tutorial. Or perhaps I'll update it. Let me look (for a week or two)

**Comment by bebatut** [comment]
Thanks @nekrut and @nsoranzo

**Pull request merged**